games/static/games/proto.js

```diff
@@ -367,7 +325,7 @@ class Board {
                 this.waitAndEraseRocketAt(rocketLocation) 
             }
             if (this.squareAt(nextRocketLocation)) {
-                if (this.squareAt(nextRocketLocation).hasAlien()) {            
+                if (this.squareAt(nextRocketLocation).checkForAlienAndRemoveIfDead()) { 
```

Comment:
it bad practice to have a function do something and return an answer
i would split it to 2:
```js
let square = this.squareAt(nextRocketLocation)
square.removeDeadAlien()
if (square.hasAlien()) {
...
}
```

Reply:
the rull of thumb is: if you have the `And` word in the function name it means the function is doing too much